tests/phpunit/tests/post/getPages.php

```diff
@@ -719,4 +773,210 @@ public function test_get_pages_cache_empty() {
+				'post_status' => 'foo',
+			)
+		);
+		$pages = get_pages(
+			array(
+				'post_status' => 'foo',
+			)
+		);
+
+		$this->assertSameSets( $posts, wp_list_pluck( $pages, 'ID' ) );
+	}
+
+	/**
+	 * @ticket 12821
+	 * @covers ::get_pages
+	 */
+	public function test_get_pages_offset() {
+		$posts = self::factory()->post->create_many( 4, array( 'post_type' => 'page' ) );
+		$pages = get_pages(
+			array(
+				'offset' => 2,
+				'number' => 2,
+			)
+		);
+
+		$this->assertSameSets( array( $posts[2], $posts[3] ), wp_list_pluck( $pages, 'ID' ) );
+	}
+
+
+	/**
+	 * @ticket 12821
+	 * @covers ::get_pages
+	 */
+	public function test_get_pages_authors() {
+		$author_1 = self::factory()->user->create(
+			array(
+				'user_login' => 'author1',
+				'role'       => 'author',
+			)
+		);
+		$post_1   = self::factory()->post->create(
+			array(
+				'post_title'  => 'Page 1',
+				'post_type'   => 'page',
+				'post_author' => $author_1,
+				'post_date'   => '2007-01-01 00:00:00',
+			)
+		);
+
+		$author_2 = self::factory()->user->create(
+			array(
+				'user_login' => 'author2',
+				'role'       => 'author',
+			)
+		);
+		$post_2   = self::factory()->post->create(
+			array(
+				'post_title'  => 'Page 2',
+				'post_type'   => 'page',
+				'post_author' => $author_2,
+				'post_date'   => '2007-01-01 00:00:00',
+			)
+		);
+		$pages    = get_pages(
+			array(
+				'authors' => "{$author_1}, {$author_2}",
+			)
+		);
+
+		$this->assertSameSets( array( $post_1, $post_2 ), wp_list_pluck( $pages, 'ID' ) );
+	}
+
+	/**
+	 * @ticket 12821
+	 * @covers ::get_pages
+	 */
+	public function test_orderby() {
+		global $wpdb;
+		// 'rand' is a valid value.
+		get_pages( array( 'sort_column' => 'rand' ) );
+		$this->assertStringContainsString( 'ORDER BY RAND()', $wpdb->last_query, 'Check order is random' );
+
+		// This isn't allowed.
+		get_pages( array( 'sort_order' => 'rand' ) );
+		$this->assertStringContainsString( 'ORDER BY', $wpdb->last_query, 'Check orderby is present' );
+		$this->assertStringNotContainsString( 'RAND()', $wpdb->last_query, 'Check order is random is not present' );
+		$this->assertStringContainsString( 'DESC', $wpdb->last_query, 'Check DESC is random is not present' );
+
+		// 'none' is a valid value.
+		get_pages( array( 'sort_column' => 'none' ) );
+		$this->assertStringNotContainsString( 'ORDER BY', $wpdb->last_query, 'Check orderby is not present' );
+		$this->assertStringNotContainsString( 'DESC', $wpdb->last_query, 'Check DESC is not present' );
+		$this->assertStringNotContainsString( 'ASC', $wpdb->last_query, 'Check ASC is not present' );
+
+		// False is a valid value.
+		get_pages( array( 'sort_column' => false ) );
+		$this->assertStringContainsString( 'ORDER BY', $wpdb->last_query, 'Check orderby is present' );
+
+		// Empty array() is a valid value.
+		get_pages( array( 'sort_column' => array() ) );
+		$this->assertStringContainsString( 'ORDER BY', $wpdb->last_query, 'Check orderby is present' );
```

Comment:
Messages will need to be different for each assertion so devs can tell which is failing if they break a test.

You can probably split this up/use a data query for each sort_column value.